R/sensititvity_page.R

```diff
@@ -499,6 +519,36 @@ SensitivityAnalysisPageServer <- function(id, data) {
       }
     )
     
+    #Run the model in the server (i.e. not inside an output) and obtain the summary
+    model_summary <- reactive({
+      newData <- data()
+      b <- study_level_outcomes(newData) # get sens, spec for each trial
+      bb <- data.frame(Author=newData$author, Year=newData$year, TP=newData$TP, FN=newData$FN, FP=newData$FP, 
+                       TN=newData$TN, N=b$N, Sens=b$Sens, Spec=b$Spec)
+      bb$Sens <- sprintf('%4.3f', bb$Sens) # restrict number of figures after decimal place for sens
+      bb$Spec <- sprintf('%4.3f', bb$Spec)
+      # add information about percentage weights 
+      N <- length(newData$TP)
+      newData$n1 <- newData$TP+newData$FN
+      newData$n0 <- newData$FP+newData$TN
+      newData$true1 <- newData$TP
+      newData$true0 <- newData$TN 
+      newData$study <- 1:N
+      Y = reshape(newData, direction = "long", varying = list( c("n1" , "n0") , c( "true1","true0" ) ) ,
+                  timevar = "sens" , times = c(1,0) , v.names = c("n","true") ) 
+      Y = Y[order(Y$id),]  
+      Y$spec<- 1-Y$sens
+      MA_Y = glmer( formula = cbind(  true , n - true ) ~ 0 + sens + spec + (0+sens + spec|study),
+                    data = Y , family = binomial , nAGQ=1 , verbose=0 ) 
+      return(summary(MA_Y))
```

Comment:
This seems to be the same code repeated again 😲 This might be a bigger job than I'd realised.